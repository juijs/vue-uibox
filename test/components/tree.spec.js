import { mount } from 'vue-test-utils'
import UixTree from '@/components/tree'

describe('tree.js', () => {
    const data = [
        { index: "0", text: "Windows" },
        { index: "0.0", text: "dll" },
        { index: "0.1", text: "zip" },
        { index: "1", text: "Download" }
    ];

    const wrapper = mount(UixTree, {
        propsData: {
            rootName: 'C:\\',
            items: data,
            item: data[2]
        }
    });

    it('snapshot!!', () => {
        wrapper.vm.$nextTick(() => {
            expect(wrapper.vm.$el).toMatchSnapshot();
        });
    });
})