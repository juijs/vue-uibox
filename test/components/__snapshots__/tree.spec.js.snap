// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tree.js snapshot!! 1`] = `
<ul
  class="tree line"
>
  <li
    class="open root"
  >
    <i />
    <div>
      <i />
       C:\\
    </div>
    <ul>
      <li
        class="open"
      >
        <i />
        <div>
          <i />
           Windows
        </div>
        <ul>
          <li
            class="leaf"
          >
            <i />
            <div>
              <i />
               dll
            </div>
            <ul />
          </li>
          <li
            class="leaf last active"
          >
            <i />
            <div>
              <i />
               zip
            </div>
            <ul />
          </li>
        </ul>
      </li>
      <li
        class="leaf last"
      >
        <i />
        <div>
          <i />
           Download
        </div>
        <ul />
      </li>
    </ul>
  </li>
</ul>
`;
